File: src/Servers/DotnetVersion/Client/BeaconTower.Client.Abstract/LogInfo.cs
using LuanNiao.JsonConverterExtends;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace BeaconTower.Client.Abstract
{
    public class LogInfo : IUseLNJsonExtends
    {
        public LogInfo()
        {
        }
        public LogInfo(LogLevel level)
        {
            Level = level;
        }

        public LogInfo SetLevel(LogLevel level)
        {
            Level = level;
            return this;
        }


        public LogInfo SetMessage(string message)
        {
            Message = message;
            return this;
        }
        public LogInfo AddCustomData(string key, string value)
        {
            CustomData.Add(key, value);
            return this;
        }
        private bool _alreadSend = false;
        public LogLevel Level { get; set; } = LogLevel.Trace;
        public long EventID { get; set; }
        public long MethodEventID { get; set; }
        public long TraceID { get; set; }
        public string Message { get; set; }
        public long MethodID { get; set; }
        public string MethodName { get; set; }
        public string FileName { get; set; }
        public int LineNumber { get; set; }
        public long TimeStamp { get; } = DateTime.Now.Ticks;
        public Dictionary<string, string> CustomData { get; } = new Dictionary<string, string>();


        public async void SendAsync()
        {
            if (_alreadSend)
            {
                return;
            }
            _alreadSend = true;
            var serverList = ServerManager.Instance.GetAvailableServer();
            var taskList = new Task[serverList.Count];
            for (int i = 0; i < serverList.Count; i++)
            {
                var index = i;
                taskList[index] = Task.Run(async () =>
                {
                    try
                    {
                        await serverList[index].Log(this);
                    }
                    catch (Exception ex)
                    {
                        Console.WriteLine($"Send message to Server: {serverList[index].Alias} failed. Message:{ex.Message}");
                    }
                });
            }

            await Task.WhenAll(taskList);
        }
    }
}

File: src/UI/src/api/model/loginfo.ts


export interface LogInfoItemResponse {
    key: string;
    level: LogLevel;
    eventID: string;
    methodEventID: string;
    traceID: string;
    message: string;
    methodID: string;
    methodName: string;
    fileName: string;
    lineNumber: number;
    timeStamp: string;
    timeInfo: string;
    customData: any;
}

export enum LogLevel {
    Trace,
    Debug,
    Info,
    Warning,
    Error,
    Panic
}

export const GetLogLevelStr = (level: LogLevel): string => {
    switch (level) {
        case LogLevel.Trace:
            return "追踪信息";
        case LogLevel.Info:
            return "提示信息";
        case LogLevel.Debug:
            return "调试信息";
        case LogLevel.Warning:
            return "警告信息";
        case LogLevel.Error:
            return "错误信息";
        case LogLevel.Panic:
            return "可能会引起崩溃";
    }
}